README.adoc

```diff
@@ -0,0 +1,180 @@
+When all of the Sections are evaluated, the resulting text is assembled into an AsciiDoc document.
+
+When creating documents for a new hardware block, we aren't creating a whole new document.
+Rather, we are creating special Sections (sometimes called "Snippets")
+which can be inserted into other documents.
+
+Scribble is expecting three Sections for each new hardware block:
+
+ - An "Overview"" paragraph describing the block.
+ - A "Programming" chapter describing how to use and program the block.
+ This section describes the registers, the memory mapped addresses,
+ and gives a detailed decription of how the block should be configured.
+ - A "HardwareInterface" chapter describing how to interface to the block.
+ This section describes the interrupts, clocks and signals used by the block.
+
+Typically, these sections will live in the "components" directory, in a subdirectory
+named after the block or device.
+
+For example, a new design for a "Whizbang" would have three files inside the following directories:
+```
+    - docs/scribble
+        - components
+            - Whizbang
+                - Overview.jinja2
+                - Programming.jinja2
+                - HardwareInterface.jinja2
+```
+
+=== What does a Section look like?
+
+A Section can be implemented as a Jinja template or as a Python function.
+In either case, it is invoked through Scribble's "Section" call.
+```
+Section("module name", scope, **context)
+```
+where:
+
+[horizontal]
+name:: is the python or jinja module name.
+scope:: is a subtree (element) of the object model.
+context:: are additional named values passed to the section.
+
+When a section is written as a Jinja template, all the
+named arguments are available as Jinja template variables.
+
+
+=== "Overview" Paragraph
+
+The overview is a simple paragraph describing the block.
+
+It is usually written as a single Jinja template.
+It doesn't need a lot of detail, but it isn't entirely static either.
+For example, if there is more than one instance of the device, then this
+section would mention how many there are.
+
+As an example, in the file `Overview.jinja2` the following line would describe how many parallel ports are in the design.
+
+----
+The {{ product_name }} contains {{ devices | length | english_number }} Parallel I/O ports.
+----
+
+which would be displayed as:
+
+----
+The PIO Test Socket contains five Parallel I/O ports.
+----
+
+=== "Programming" Chapter
+This chapter describes the device or block and how to use it.
+For some devices, the Section can get quite large and include a hierarchy
+of sections and subsections. For simpler blocks, it might be a single Jinja
+template file named `Programming.jinja2`.
+
+
+=== "Hardware Interface" Chapter
+The hardware interface chapter decribes how the block interfaces with the rest of the design. It would
+describe the input and output signals, how they are clocked, and what interrupts the device generates.
```

Comment:
Interrupts belong in the Programming chapter.

Also, in general, the voice is too passive, and we should make this a lot more prescriptive. Rather than using words like "would" or "might", we should favor words like "should" or "is". As a reader of this README, I want to know what I should do to comply with the standard, and reading passive language like this makes it less whether these are suggestions or requirements.